src/Yaapii.Atoms/Scalar/And.cs

```diff
@@ -2,147 +2,133 @@
-        /// <param name="funcs">the conditions to apply</param>
-        public And(EnumerableOf<System.Func<bool>> funcs) : this(
-            new Enumerable.Mapped<System.Func<bool>, IScalar<bool>>(
+        private And(And and)
+        {
+            _and = and;
+        }
+
+        /// <summary> Get the value. </summary>
+        /// <returns> the value </returns>
+        public Boolean Value()
+        {
+            return _and.Value();
+        }
+    }
+
+    /// <summary> Logical and. Returns true if all contents return true. </summary>
+    public sealed class And : IScalar<bool>
+    {
+        private readonly IEnumerable<IScalar<bool>> _enumerable;
+
+        /// <summary> Logical and. Returns true if all calls to <see cref="Func{In, Out}"/> were true. </summary>
+        /// <param name="funcs"> the conditions to apply </param>
+        public And(params Func<bool>[] funcs) : this(new EnumerableOf<System.Func<bool>>(funcs))
+        { }
+
+        /// <summary> Logical and. Returns true if all calls to <see cref="Func{Out}"/> were true. </summary>
+        /// <param name="funcs"> the conditions to apply </param>
+        public And(EnumerableOf<Func<bool>> funcs) : this(
+            new Mapped<Func<bool>, IScalar<bool>>(
                 func => new ScalarOf<bool>(func),
                 funcs))
         { }
 
-        /// <summary>
-        /// ctor
-        /// </summary>
-        /// <param name="src">list of items</param>
+        /// <summary> ctor </summary>
+        /// <param name="src"> list of items </param>
         public And(params IScalar<Boolean>[] src) : this(
             new EnumerableOf<IScalar<Boolean>>(src))
         { }
 
-        /// <summary>
-        /// ctor
-        /// </summary>
-        /// <param name="src">list of items</param>
+        /// <summary> ctor </summary>
+        /// <param name="src"> list of items </param>
+        public And(params bool[] src) : this(
+            new Mapped<bool, IScalar<bool>>(
+                tBool => new ScalarOf<bool>(tBool),
+                src))
+        { }
+
+        /// <summary> ctor </summary>
+        /// <param name="src"> list of items </param>
+        public And(IEnumerable<bool> src) : this(
+            new Mapped<bool, IScalar<bool>>(
+                tBool => new ScalarOf<bool>(tBool),
+                src))
+        { }
+
+        /// <summary> ctor </summary>
+        /// <param name="src"> list of items </param>
         public And(IEnumerable<IScalar<Boolean>> src)
         {
-            this._enumerable = src;
+            _enumerable = src;
         }
 
-        /// <summary>
-        /// Get the value.
-        /// </summary>
-        /// <returns>the value</returns>
-        public Boolean Value()
+        /// <summary> Get the value. </summary>
+        /// <returns> the value </returns>
+        public bool Value()
         {
             Boolean result = true;
-            foreach (IScalar<Boolean> item in this._enumerable)
```

Comment:
do not remove this, we want this to be present

Reply:
I dont think we need the 'this' to notice using a class variable, because our class variables are marked with '_'. I uses 'this' only if it is needed. It is like write for every methode full namespace (in my opinion)

Reply:
You are right, this and underscore are redundant information. Personally I like using "this" better (which has changed over time). Underscore looks more technical than "this" and also "this" is more declarative. So I think about getting rid of the underscore in the future and that requires the this.